### PR TITLE
refactor(react-grid): extract table column extension

### PR DESCRIPTION
**Pull request by kvet**
BREAKING CHANGES:

In order to improve API readability, we extracted the `width` and `align` column properties that the Table and VirtualTable plugins handle into a separate property. Now, the `width` and `align` properties in the Grid `columns` are no longer supported. Use a TableColumnExtension instead:

Before:

```js
const columns= [{ name: 'field', width: 100 }, { name: 'field2' }];

<Grid columns={columns}>
  <Table />
</Grid>
```

After:

```js
const columns= [{ name: 'field1' }, { name: 'field2' }];
const tableColumnExtensions = [{ columnName: 'field1', width: 100 }];

<Grid columns={columns}>
  <Table columnExtensions={tableColumnExtensions} />
</Grid>
```

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/table/computeds.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-export const tableColumnsWithDataRows = columns =>
-  columns.map(column => ({
-    key: `${TABLE_DATA_TYPE}_${column.name}`,
-    type: TABLE_DATA_TYPE,
-    width: column.width,
-    column,
-  }));
+export const tableColumnsWithDataRows = (columns, columnExtensions) =>
+  columns.map((column) => {
+    const columnExtension = getColumnExtension(columnExtensions, column.name);
```
Comment (SergeyAlexeev) [inline packages/dx-grid-core/src/plugins/table/computeds.js]: The `column.name` field is used twice and can be extracted.

File: packages/dx-grid-core/src/plugins/table/computeds.test.js [hunk 2]
```diff
@@ -16,12 +16,21 @@ describe('Table Plugin computeds', () => {
...
         .toMatchObject({ width: 100 });
     });
+
+    it('should set align from columnExtension', () => {
+      const columns = [{ name: 'b' }];
+      const columnExtensions = [{ columnName: 'b', align: 'right' }];
+
+      expect(tableColumnsWithDataRows(columns, columnExtensions)[0])
+        .toMatchObject({ align: 'right' });
+    });
```
Comment (SergeyAlexeev) [inline packages/dx-grid-core/src/plugins/table/computeds.test.js]: Should we add one more test that checks that only the `width` and `align` fields will be applied?

File: packages/dx-react-demos/src/material-ui/featured-controlled-mode/demo.jsx [hunk 3]
```diff
@@ -200,14 +187,16 @@ class DemoBase extends React.PureComponent {
...
+        { name: 'region', title: 'Region' },
+        { name: 'amount', title: 'Sale Amount' },
+        { name: 'discount', title: 'Discount' },
         { name: 'saleDate', title: 'Sale Date' },
         { name: 'customer', title: 'Customer' },
       ],
+      tableColumnExtensions: [
+        { columnName: 'amount', align: 'right', cellComponent: HighlightedCell },
+        { columnName: 'discount', cellComponent: ProgressBarCell },
+      ],
```
Comment (SergeyAlexeev) [inline packages/dx-react-demos/src/material-ui/featured-controlled-mode/demo.jsx]: Why there isn't the `width` extension as in the prev version?

File: packages/dx-grid-core/src/plugins/table/computeds.test.js (hunk 2 above)
Comment (kvet) [reply packages/dx-grid-core/src/plugins/table/computeds.test.js]: No, I think that these tests are enough

**Pull request merged**